lib/src/components/NxNavigationSidebar/types.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { ValidationMap, ReactNode, HTMLAttributes, Validator } from 'react';
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+
+export interface Props {
+  isOpen: boolean;
+  className?: string | null;
+  toggleOpenIcon: IconDefinition;
+  toggleCloseIcon: IconDefinition;
+  onToggleClick: (() => void);
+  logoImg?: string | null;
+  logoText?: string | null;
+  logoLink: string;
+}
+
+export const propTypes: ValidationMap<Props> = {
+  isOpen: PropTypes.bool.isRequired,
+  className: PropTypes.string,
+  toggleOpenIcon: PropTypes.object.isRequired as Validator<IconDefinition>,
+  toggleCloseIcon: PropTypes.object.isRequired as Validator<IconDefinition>,
+  onToggleClick: PropTypes.func.isRequired,
+  logoImg: PropTypes.string,
+  logoText: PropTypes.string,
+  logoLink: PropTypes.string.isRequired
+};
+
+export type NxNavigationSidebarContentProps = HTMLAttributes<HTMLDivElement> & {
```

Comment:
I don't think there's value in this type.  `NxNavigationSidebarContent` takes the same props as a `<section>`.

Reply:
> I don't think there's value in this type. `NxNavigationSidebarContent` takes the same props as a `<section>`.

I was thinking something like:

```
HTMLAttributes<HTMLSectionElement> & {
  children: ReactNode | ReactNode[];
};
```

But I don't see any `HTMLSectionElement` definition that I could use. Isn't `HTMLDivElement` the next most similar element? Or I don't understand your objection to this.

Reply:
Just to add 2¢, you shouldn't have to do `children: ReactNode | ReactNode[];`, the `ReactNode` type includes `JSX.Element | JSX.Element[]` so `children: ReactNode` would be fine

Reply:
> Or I don't understand your objection to this.

What I'm saying is get rid of the `NxNavigationSidebarContentProps` entirely and just specify the type of `NxNavigationSidebarLinks` as `FunctionComponent<HTMLAttributes<HTMLDivElement>>`.  Your extra specification of the `children` is effectively just making the `children` required instead of optional.  Which is of very little value.

Reply:
> > Or I don't understand your objection to this.
> 
> What I'm saying is get rid of the `NxNavigationSidebarContentProps` entirely and just specify the type of `NxNavigationSidebarLinks` as `FunctionComponent<HTMLAttributes<HTMLDivElement>>`. Your extra specification of the `children` is effectively just making the `children` required instead of optional. Which is of very little value.

I couldn't remove the propTypes entirely cuz then the linter started to complain about `className` not being defined despite the fact that propTypes is NOT defining className at the moment and is getting it from the inherited type. I really dislike this type of typing issues.

So I just simplified the type and left the propTypes with optional children. ba0b3b26b4170f3b8360e7e342c33b8936f011e2

I welcome any eyes and/or suggestions to further simplify this.

Reply:
> I welcome any eyes and/or suggestions to further simplify this.

Given that `className` is the only prop that you're explicitly working with, I think it would make sense for _that_ to be the one in the propTypes.  It is weird that it's not complaining about it when you have a propType that only defines the `children` prop.

Reply:
> > I welcome any eyes and/or suggestions to further simplify this.
> 
> Given that `className` is the only prop that you're explicitly working with, I think it would make sense for _that_ to be the one in the propTypes. It is weird that it's not complaining about it when you have a propType that only defines the `children` prop.

Done in ad6f86ea49a86f06ae0b269e8a5844b90acbca1c